src/utils/fiatToCrypto.ts

```diff
@@ -0,0 +1,187 @@
+    optionStr += `${!optionStr ? '?' : '&'}${key}=${encodeURIComponent(value)}`;
+  });
+  return optionStr;
+};
+
+// RAMP
+export const rampWidgetUrl = (
+  address: string,
+  fiatCurrency: string = '',
+  fiatValue: string = '',
+  isEtherspotAccount: boolean = true,
+) => {
+  const params = {
+    hostAppName: PILLAR,
+    hostApiKey: getEnv().RAMPNETWORK_API_KEY,
+    userAddress: address,
+    fiatCurrency,
+    fiatValue,
+    swapAsset: isEtherspotAccount ? ETHERSPOT_RAMP_CURRENCY_TOKENS.join(',') : ARCHANOVA_RAMP_CURRENCY_TOKENS,
+  };
+
+  const url = `${getEnv().RAMPNETWORK_WIDGET_URL}/${buildUrlOptions(params)}`;
+  return url;
+};
+
+// WERT
+export const wertWidgetUrl = (address: string, fiatValue: string) => {
+  const wertWidget = new WertWidget({
+    partner_id: getEnv().WERT_ID,
+    container_id: CONTAINER_ID,
+    origin: ORIGIN,
+    commodities: firebaseRemoteConfig.getString(REMOTE_CONFIG.FEATURE_WERT_COMMODITIES),
+    currency: firebaseRemoteConfig.getString(REMOTE_CONFIG.FEATURE_WERT_CURRENCY),
+    currency_amount: parseFloat(fiatValue),
+    commodity: firebaseRemoteConfig.getString(REMOTE_CONFIG.FEATURE_WERT_COMMODITY),
+    address,
+  });
+
+  return wertWidget.getEmbedUrl();
+};
+
+// Pelerin
+export const buildMtPelerinOptions = (code: string, address: string) => {
+  let onRampOptions = {
+    lang: 'en',
+    tab: 'buy',
+    tabs: 'buy',
+    chain: 'xdai_mainnet',
+    net: 'matic_mainnet',
+    nets: 'arbitrum_mainnet,avalanche_mainnet,bsc_mainnet,fantom_mainnet,mainnet,optimism_mainnet,xdai_mainnet',
+    crys: 'AVAX,BNB,BTCB,BUSD,DAI,ETH,FRAX,LUSD,MAI,MATIC,RBTC,RDOC,RIF,USDC,USDT,WBTC,WETH,XCHF,XDAI,XTZ',
+    rfr: 'etherspot',
+    bsc: 'GBP',
+    bdc: 'MATIC',
+    hash: '',
+    code: code,
+    addr: address || '',
+  };
+
+  return onRampOptions;
+};
+
+export const pelerinWidgetUrl = async (
+  deployingAccount = false,
+  setDeployingAccount?: (value: boolean) => void,
+  showAlert?: (message: string) => void,
+) => {
+  const chain = chainFromChainId[CHAIN_ID.XDAI];
+
+  const sdk: Sdk | undefined = etherspotService.getSdkForChain(chain);
+  const account: Account | undefined = await etherspotService.getAccount(chain);
+
+  let base64Hash = '';
+  const code = Math.floor(Math.random() * 8999) + 1000;
+  const message = 'MtPelerin-' + code;
+  let onRampOptions = buildMtPelerinOptions(code.toString(), account.address);
+
+  if (account?.state === AccountStates.UnDeployed) {
+    if (deployingAccount) {
+      !!showAlert && showAlert('Deploying Etherspot wallet, please try again later.');
+      return;
+    }
+
+    !!setDeployingAccount && setDeployingAccount(true);
+    const hash: string = await etherspotService.setBatchDeployAccount(chain, true);
+    !!setDeployingAccount && setDeployingAccount(false);
+
+    if (!hash || hash === AccountStates.Deployed || hash === AccountStates.UnDeployed) {
+      !!showAlert && showAlert('Failed to deploy the Etherspot wallet');
```

Comment:
Please add a suggestion what to do next after the message, such as try again later. This string should also be in the i18n files.